fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/OrExpansion.java

```diff
@@ -73,63 +80,109 @@ public class OrExpansion extends OneExplorationRuleFactory {
             .build();
 
     @Override
-    public Rule build() {
-        return logicalJoin(any(), any()).when(JoinUtils::shouldNestedLoopJoin)
-                .whenNot(LogicalJoin::isMarkJoin)
-                .when(join -> supportJoinType.contains(join.getJoinType())
-                        && ConnectContext.get().getSessionVariable().getEnablePipelineEngine())
-                .thenApply(ctx -> {
-                    LogicalJoin<? extends Plan, ? extends Plan> join = ctx.root;
-                    Preconditions.checkArgument(join.getHashJoinConjuncts().isEmpty(),
-                            "Only Expansion nest loop join without hashCond");
+    public Plan rewriteRoot(Plan plan, JobContext jobContext) {
+        OrExpandsionContext ctx = new OrExpandsionContext(
+                jobContext.getCascadesContext().getStatementContext(), jobContext.getCascadesContext());
+        if (!(plan instanceof LogicalResultSink)) {
```

Comment:
if we have cte in sql, the root may not be LogicalResultSink. the plan could be
```
CteAnchor1
+-- CteProducer1
+-- CteAnchor2
    +-- CteProducer2
    +-- ResultSink
        +-- ...
```

Reply:
BTW, u should use `LogicalSink` to support insert into

Reply:
fixed, add CTEAnchor above Sink